docs/parse_argparse.py

```diff
@@ -0,0 +1,315 @@
+    These rst files will be called by the toctreee in scripts_main.rst
+
+    Parameters
+    ----------
+    fpath : string
+        file path that needs to be documented,
+        e.g. '</path/to/autometa/script.py>'
+    rst_dirpath : string
+        path to the directory where script.rst files will be written,
+        e.g. '</path/to/docs/source/scripts/directory>'
+    """
+    # extract "/home/siddharth/Autometa/autometa/common/external"
+    dir_path = os.path.dirname(os.path.abspath(fpath))
+    rst_main_dir = os.path.basename(dir_path)  # extraxt "external"
+    rst_main_file = rst_main_dir + "_main.rst"  # making a file external_main
+    path_main_file = os.path.join(rst_dirpath, rst_main_file)
+    outlines = ""
+    if not os.path.exists(path_main_file):
+        with open(path_main_file, "w") as fh:
+            lines = "\n".join(["="*len(rst_main_dir), rst_main_dir, "="*len(rst_main_dir), " ", ".. toctree::",
+                               "\t :maxdepth: 2", "\t :caption: Table of Contents", " "])
+            outlines += lines
+            path_recorded = ""
+            for __, dirnames, filename in os.walk(dir_path):
+                # This is done to prevent writing directories below  the sub-directories
+                # i.e. we don't need to link any directories (if they are in future) that are below external with the common toctree
+                # the way this `if` statements works is: os.walk will keep iterating through a directory recursively (going into sub-sub-directories), we just need its first output
+                # of the directories within the root, after that it goes through the sub-sub-directories and lists their directories which we don't want,
+                # meanwhile path_main_file is constant, thus we can skip recording sub-subdirectories if we tell it that, this path has alreasy been recorded.
+                # remember path_main_file is remaining constant for one dir_path
+                if path_recorded == path_main_file:
+                    continue
+                # this counts the number of sub-directories each directory has
+                # true only for `common` (as of April 24, 2020)
+                if len(dirnames) > 0:
+                    for dirname in dirnames:
+                        if "__" in dirname:  # to prevent __pychache__ folders from being considered
+                            continue
+                        dir_main = dirname + "_main"  # external_main
+                        path_recorded = path_main_file
+                        lines = f"\n\t {dirname}/{dir_main}"
+                        outlines += lines
+            fh.write(outlines)
+    # Opens each DirectoryName_main.rst and add the name of the scripts in that directory
+    with open(path_main_file, "a") as fh:
+        fname = os.path.basename(fpath)
+        # extract the "kmers" of kmers.py
+        basename = os.path.splitext(fname)[0]
+        fh.write(f"\n\t {basename}")
```

Comment:
This is a confusing variable name (`basename`), because you are calling the basename method above then doing something else and calling this variable basename.

```suggestion
        basename = os.path.basename(fpath)
        # extract the "kmers" of kmers.py
        fname = os.path.splitext(basename)[0]
        fh.write(f"\n\t {fname}")
```